src/features/integration/IntegrationConfigurationForm.tsx

```diff
@@ -0,0 +1,143 @@
+import ExpandMoreIcon from '@mui/icons-material/ExpandMore';
+import RecordForm from "./components/form/RecordForm";
+import DocumentForm from "./components/form/DocumentForm";
+import ApplicantForm from "./components/form/ApplicantForm";
+import CaseForm from "./components/form/CaseForm";
+import IFormData from "./types/Form/FormData";
+import CaseInformation from "./components/form/CaseInformation";
+import IntegrationService from "./service/IntegrationService";
+import {defaultValues} from "./util/DefaultValues";
+import {mapToDto} from "./util/MapToDto";
+
+const useStyles = makeStyles((theme: Theme) =>
+    createStyles({
+        form: {
+            width: 1200
+        },
+        formControl: {
+            width: 750
+        },
+        accordion: {
+            marginBottom: theme.spacing(2)
+        },
+        button: {
+            marginRight: theme.spacing(1)
+        },
+        submitButton: {
+            backgroundColor: theme.palette.primary.dark,
+            border: 'none',
+            color: 'white',
+            padding: theme.spacing(2),
+            cursor: 'pointer'
+        }
+    }));
+
+const IntegrationConfigurationForm: React.FunctionComponent<RouteComponentProps<any>> = () => {
+    const classes = useStyles();
+    const [expanded, setExpanded] = React.useState<string | false>(false);
+    const handleChange =
+        (panel: string) => (event: React.SyntheticEvent, isExpanded: boolean) => {
+            setExpanded(isExpanded ? panel : false);
+        };
+
+    const {handleSubmit, watch, setValue, formState: {}} = useForm<IFormData>({
+        defaultValues: defaultValues
+    });
+
+    const createNewConfiguration = (data: any) => {
+        IntegrationService.create(data)
+            .then(response => {
+                console.log('created new configuraton', data);
+            })
+            .catch((e: Error) => {
+                console.log('errror creating new', e);
+            });
+    }
+
+    const updateConfiguration = (data: any, id: any) =>{
+        IntegrationService.update(id, data)
+            .then(response => {
+                console.log('updated configuration', id);
+            })
+            .catch((e: Error) => {
+                console.log('errror updating',e);
+            })
+    }
+
+    const onSubmit = handleSubmit((data: IFormData) => {
+        const dto = mapToDto(data);
+
+        if(data.id) {
+            updateConfiguration(dto, data.id);
+        } else {
+            createNewConfiguration(dto);
+        }
+    });
+
+    return (
+        <Box display="flex" position="relative" width={1} height={1}>
+            <Box>
+                <Typography variant={"h5"} sx={{mb: 2}}>Integrasjonskonfigurasjon</Typography>
+                <form className={classes.form} onSubmit={onSubmit}>
+                    <Accordion expanded={expanded === 'panel1'} onChange={handleChange('panel1')} className={classes.accordion} defaultExpanded={true}>
```

Comment:
Can this be simplified? Create a component? There is a lot of simular setup of each of them.

Reply:
added a new accordioncompoenent